snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/internal/utils/Util.java

```diff
@@ -305,87 +308,53 @@ static public SelfDescribingJson getApplicationContext(@NonNull Context context)
     // --- Mobile Context
 
     /**
-     * Returns the Mobile Context
-     *
-     * @param context the Android context
-     * @return the mobile context
-     */
-    @Nullable
-    public static SelfDescribingJson getMobileContext(@NonNull Context context) {
-        Map<String, Object> pairs = new HashMap<>();
-        addToMap(Parameters.OS_TYPE, getOsType(), pairs);
-        addToMap(Parameters.OS_VERSION, getOsVersion(), pairs);
-        addToMap(Parameters.DEVICE_MODEL, getDeviceModel(), pairs);
-        addToMap(Parameters.DEVICE_MANUFACTURER, getDeviceVendor(), pairs);
-        addToMap(Parameters.CARRIER, getCarrier(context), pairs);
-        addToMap(Parameters.ANDROID_IDFA, getAndroidIdfa(context), pairs);
-
-        NetworkInfo networkInfo = getNetworkInfo(context);
-        addToMap(Parameters.NETWORK_TYPE, getNetworkType(networkInfo), pairs);
-        addToMap(Parameters.NETWORK_TECHNOLOGY, getNetworkTechnology(networkInfo), pairs);
-
-        if (mapHasKeys(pairs,
-                Parameters.OS_TYPE,
-                Parameters.OS_VERSION,
-                Parameters.DEVICE_MANUFACTURER,
-                Parameters.DEVICE_MODEL)) {
-            return new SelfDescribingJson(TrackerConstants.MOBILE_SCHEMA, pairs);
-        } else {
-            return null;
-        }
-    }
-
-    /**
+     * @deprecated Moved to internal API.
      * @return the OS Type
      */
     @NonNull
     public static String getOsType() {
```

Comment:
I'm not sure that the `@deprecated` tag in the Javadoc is enough to inform the compiler.
I guess we should add also the `@Deprecated` annotation in the signature in order to inform the compiler and raise the warning at build time.

Also, for the doc, we inform that we moved it.
What if we just say that it will be removed in the v3?
We don't want they use the internal classes, so in case they complain we can decide to leave these methods in the public API for v3 too.